src/python/pants/option/config.py

```diff
@@ -263,35 +267,35 @@ class _ChainedConfig(Config):
   chained_configs: Tuple[Config, ...]
 
   @property
-  def _configs(self):
+  def _configs(self) -> Tuple[Config, ...]:
     return self.chained_configs
 
-  def configs(self):
+  def configs(self) -> Tuple[Config, ...]:
     return self.chained_configs
 
-  def sources(self):
+  def sources(self) -> List[str]:
     # NB: Present the sources in the order we were given them.
     return list(itertools.chain.from_iterable(cfg.sources() for cfg in reversed(self._configs)))
 
-  def sections(self):
+  def sections(self) -> List[str]:
     ret = OrderedSet()
     for cfg in self._configs:
       ret.update(cfg.sections())
-    return ret
+    return list(ret)
```

Comment:
The superclass and two sibling classes were returning a `List`, so I changed it here to do the same.